ReactWindows/ReactNative/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,258 @@
+            {
+                return false;
+            }
+
+            bool result;
+            switch (uiElementAutomationPeer.Navigate(AutomationNavigationDirection.Parent))
+            {
+                case FrameworkElementAutomationPeer parentFEAutomationPeer:
+                    if (parentFEAutomationPeer.Owner != null)
+                    {
+                        var importantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentFEAutomationPeer.Owner);
+                        if (importantForAccessibilityAttached == ImportantForAccessibility.Yes ||
+                            importantForAccessibilityAttached == ImportantForAccessibility.NoHideDescendants)
+                        {
+                            return true;
+                        }
+                    }
+                    result = IsHiddenByAncestor(parentFEAutomationPeer);
+                    break;
+                case ItemAutomationPeer parentIAutomationPeer:
+                    if (parentIAutomationPeer.Item is UIElement parentUIElement)
+                    {
+                        var importantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentUIElement);
+                        if (importantForAccessibilityAttached == ImportantForAccessibility.Yes ||
+                            importantForAccessibilityAttached == ImportantForAccessibility.NoHideDescendants)
+                        {
+                            return true;
+                        }
+                    }
+                    result = IsHiddenByAncestor(parentIAutomationPeer);
+                    break;
+                case AutomationPeer parentAutomationPeer:
+                    result = IsHiddenByAncestor(parentAutomationPeer);
+                    break;
+                case null:
+                    result = false;
+                    break;
+                default:
+                    result = false;
+                    break;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Recursively sets the value of AccessibilityView according ImportantForAccessibility setting
+        /// for children of <paramref name="uiElementAutomationPeer"/>
+        /// </summary>
+        /// <param name="uiElementAutomationPeer"></param>
+        private static void SetChildrenAccessibilityViewFromImportantForAccessibility(AutomationPeer uiElementAutomationPeer)
+        {
+            if (uiElementAutomationPeer?.GetChildren() == null)
+            {
+                return;
+            }
+
+            foreach (AutomationPeer automationPeer in uiElementAutomationPeer.GetChildren())
+            {
+                UIElement childUIElement = GetUIElementFromAutomationPeer(automationPeer);
+                if (childUIElement == null)
+                {
+                    SetChildrenAccessibilityViewFromImportantForAccessibility(automationPeer);
+                    continue;
```

Comment:
>continue [](start = 20, length = 8)

Instead of using `continue`, should we just have an `else` branch? In cases like this where an `if/else` could work, I think using constructs like `continue` or `break` make the code harder to read because they cause your control flow to jump around.

Reply:
I agree it is better in most cases to have one exit point for readability. 'Continue' is however often an exception, many because we want to avoid big ifs and unnecessary indentation. In this case since if is simple enough, I'll change it, but if we had more code after the if I would keep the continue in place.


---
In reply to: [164943576](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r164943576) [](ancestors = 164943576)